packages/core/js-client/src/index.ts

```diff
@@ -13,157 +13,236 @@
-     * Handle connection state changes. Immediately returns the current connection state
-     */
-    onConnectionStateChange(handler: (state: ConnectionState) => void): ConnectionState {
-        return this.defaultClient?.onConnectionStateChange(handler) || 'disconnected';
-    },
+  /**
+   * Disconnect from the Fluence network
+   */
+  disconnect: async function (): Promise<void> {
+    await this.defaultClient?.disconnect();
+    this.defaultClient = undefined;
+  },
 
-    /**
-     * Low level API. Get the underlying client instance which holds the connection to the network
-     * @returns IFluenceClient instance
-     */
-    getClient: async function(): Promise<IFluenceClient> {
-        if (!this.defaultClient) {
-            throw new Error('Fluence client is not initialized. Call Fluence.connect() first');
-        }
-        return this.defaultClient;
-    },
+  /**
+   * Handle connection state changes. Immediately returns the current connection state
+   */
+  onConnectionStateChange(handler) {
+    return (
+      this.defaultClient?.onConnectionStateChange(handler) ?? "disconnected"
+    );
+  },
+
+  /**
+   * Low level API. Get the underlying client instance which holds the connection to the network
+   * @returns IFluenceClient instance
+   */
+  getClient: function () {
+    if (this.defaultClient == null) {
+      throw new Error(
+        "Fluence client is not initialized. Call Fluence.connect() first",
+      );
+    }
+
+    return this.defaultClient;
+  },
 };
 
-export type { IFluenceClient, ClientConfig, CallParams } from '@fluencelabs/interfaces';
+export type {
+  IFluenceClient,
+  ClientConfig,
+  CallParams,
+} from "@fluencelabs/interfaces";
 
 export type {
-    ArrayType,
-    ArrowType,
-    ArrowWithCallbacks,
-    ArrowWithoutCallbacks,
-    BottomType,
-    FunctionCallConstants,
-    FunctionCallDef,
-    LabeledProductType,
-    NilType,
-    NonArrowType,
-    OptionType,
-    ProductType,
-    ScalarNames,
-    ScalarType,
-    ServiceDef,
-    StructType,
-    TopType,
-    UnlabeledProductType,
-    CallAquaFunctionType,
-    CallAquaFunctionArgs,
-    PassedArgs,
-    FnConfig,
-    RegisterServiceType,
-    RegisterServiceArgs,
-} from '@fluencelabs/interfaces';
-
-export { v5_callFunction, v5_registerService } from './api.js';
-
-// @ts-ignore
+  ArrayType,
+  ArrowType,
+  ArrowWithCallbacks,
+  ArrowWithoutCallbacks,
+  BottomType,
+  FunctionCallConstants,
+  FunctionCallDef,
+  LabeledProductType,
+  NilType,
+  NonArrowType,
+  OptionType,
+  ProductType,
+  ScalarNames,
+  ScalarType,
+  ServiceDef,
+  StructType,
+  TopType,
+  UnlabeledProductType,
+  CallAquaFunctionType,
+  CallAquaFunctionArgs,
+  PassedArgs,
+  FnConfig,
+  RegisterServiceType,
+  RegisterServiceArgs,
+} from "@fluencelabs/interfaces";
+
+export { v5_callFunction, v5_registerService } from "./api.js";
+
+// @ts-expect-error Writing to global object like this prohibited by ts
 globalThis.new_fluence = Fluence;
```

Comment:
Don't know which one is better
```suggestion
declare global {
  // eslint-disable-next-line no-var
  var new_fluence: FluencePublicApi;
}

globalThis.new_fluence = Fluence;
```

Reply:
Looks the same to me